pkg/token/runtimetokenprovider/methods.go

```diff
@@ -0,0 +1,162 @@
+
+	providerURL, err := p.runtimeTokenProvider.BuildProviderURL(pkg.SpiffeTokenProviderGetTokenAPI)
+	if err != nil {
+		return "", fmt.Errorf("failed to build token provider URL: %w", err)
+	}
+
+	// requiredSecrets are runtime created and is provided from the configuration
+	requiredSecrets := strings.Split(p.runtimeTokenProvider.RequiredSecrets, ",")
+
+	form := url.Values{
+		"service_key":        []string{serviceKey},
+		"known_secret_names": requiredSecrets,
+		"raw_token":          []string{"true"},
+	}
+	formVal := form.Encode()
+	req, err := http.NewRequest(http.MethodPost, providerURL, strings.NewReader(formVal))
+	if err != nil {
+		return "", fmt.Errorf("failed to prepare spiffe-token-provider gettoken request: %w", err)
+	}
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("failed to send spiffe-token-provider gettoken request: %w", err)
+	}
+	defer resp.Body.Close()
+	responseBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	var rawToken string
+	var gotTokenErr error
+	switch resp.StatusCode {
+	case http.StatusOK:
+		rawToken = string(responseBody)
+		p.lc.Info("successfully got token from spiffe-token-provider!")
+	default:
+		gotTokenErr = fmt.Errorf("failed to get spiffe-token-provider gettoken api call, status code = %d ", resp.StatusCode)
+	}
+
+	return rawToken, gotTokenErr
+}
+
+func (p *runtimetokenprovider) SetTLSConfigFunc(tlsConfFunc func(context.Context, logger.LoggingClient, string) (*tls.Config, error)) {
+	p.getTLSConfig = tlsConfFunc
+}
+
+func (p *runtimetokenprovider) defaultGetTLSConfig(ctx context.Context, lc logger.LoggingClient, socketPath string) (*tls.Config, error) {
+	udsSocket := socketPath
+	if !strings.HasPrefix(socketPath, "unix://") {
+		udsSocket = "unix://" + socketPath
+	}
+
+	lc.Infof("using Unix Domain Socket at %s", udsSocket)
+
+	// command-line equivalent: spire-agent api fetch x509 -socketPath xxxx
+	// Create a `workloadapi.X509Source`, it will connect to Workload API using provided socket path
+	source, err := workloadapi.NewX509Source(ctx, workloadapi.WithClientOptions(workloadapi.WithAddr(udsSocket)))
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create X509Source: %v", err)
+	}
+
+	// need to cache the source so that we can close it later after https TLS calls
+	p.source = source
+
+	lc.Info("workload got X509 source")
+
+	trustDomainString := os.Getenv("SPIFFE_TRUST_DOMAIN")
+	if len(trustDomainString) == 0 {
+		return nil, fmt.Errorf("environment variable SPIFFE_TRUST_DOMAIN must be set")
+	}
+
+	td, err := spiffeid.TrustDomainFromString(trustDomainString)
+	if err != nil {
+		return nil, fmt.Errorf("could not get SPIFFE trust domain from string '%s': %v", trustDomainString, err)
+	}
+
+	tlsConf := tlsconfig.MTLSClientConfig(source, source, tlsconfig.AuthorizeMemberOf(td))
```

Comment:
Use `MTLSWebClientConfig` instead.  The client doesn't need to assert that the server is part of the SPIFFE_TRUST_DOMAIN.  We only need to verify the server against its certificate bundle.  Can get rid of the Getenv and TrustDomainFromString calls as a result.